src/instructlab/configuration.py

```diff
@@ -85,6 +85,18 @@ class _general(BaseModel):
         default=False,
         description="Use legacy IBM Granite chat template (default uses 3.0 Instruct template)",
     )
+    # Global student model id
+    student_model_id: str | None = Field(
```

Comment:
Try using `typing.NotRequired` here, we should see whether this type can allow us to avoid explicitly set this field within the config

```suggestion
    student_model_id: typing.NotRequired[str] = Field(
```

Reply:
Specifically today all config fields will be rendered in the default on-disk config which is super confusing. I want to see if using this type will prevent this field from being listed there entirely until it's necessary.

Reply:
@jaideepr97 I see, thank you for checking this. The way you have done it is perfect